modules/ipp/scanner.go

```diff
@@ -566,49 +551,60 @@ func (scanner *Scanner) newIPPScan(target *zgrab2.ScanTarget) *scan {
 	return &newScan
 }
 
+// TODO: Do you want to retry with TLS for all versions? Just one's you've already tried? Haven't tried? Just the same version?
+func (scanner *Scanner) tryGrabForVersions(target *zgrab2.ScanTarget, versions *[]version) (*scan, *zgrab2.ScanError) {
+	scan := scanner.newIPPScan(target)
+	// TODO: Implement scan.Cleanup()
+	var err *zgrab2.ScanError
+	for i := 0; i < len(*versions); i++ {
+		err = scanner.Grab(scan, target, &(*versions)[i])
+		if err != nil && err.Err == ErrVersionNotSupported && i < len(*versions)-1 {
+			continue
```

Comment:
This construction is a little weird -- I'm guessing the reasoning for `if (a) { continue } break` rather than `if !a { break }` is `!a` is kind of clumsy?